src/borg/remote.py

```diff
@@ -755,8 +766,11 @@ def handle_error(unpacked):
                             raise UnexpectedRPCDataFormatFromServer(data)
                         if msgid in self.ignore_responses:
                             self.ignore_responses.remove(msgid)
+                            # async methods never return values, but may raise exceptions.
                             if b'exception_class' in unpacked:
-                                handle_error(unpacked)
```

Comment:
Do we really want to arbitrarily delay error handling and grow self.async_responses without small bound here? That seems like a problem in cases where we do not plan to ignore the errors anyway.
Does this need to be parameterised? Do we need to ensure that the errors are only delivered in the barrier() for the repository fixing commands, or is the important part just that the current command is executed and no errors occur after the barrier() call?

Thinking about this... Barrier currently conflates two different parts. It ensures that all outstanding calls are finished and it does retrieve the accumulated errors. Maybe have barrier() just wait and have a lightweight call that retrieves the async errors. So this would allow retrieving errors without delay as soon as they are seen, but does not disturb the current command?